src/vsphere_cpi/spec/unit/cloud/vsphere/vm_placement_selection_pipeline_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'spec_helper'
+require 'cloud/vsphere/disk_placement_selection_pipeline'
+require 'cloud/vsphere/resources/datastore'
+
+describe VSphereCloud::VmPlacementSelectionPipeline do
+  subject { described_class.new(*criteria) {[placement_1, placement_2]} }
+
+  def fake_placement(cluster)
+    VSphereCloud::VmPlacement.new(cluster: cluster, datastores: cluster.accessible_datastores.values, hosts: nil)
+  end
+
+
+  def fake_datastore(name, free_space: 4096)
+    VSphereCloud::Resources::Datastore.new(
+      name, true, instance_double('VimSdk::Vim::Datastore'), 8192, free_space
+    ).tap do |resource|
+      allow(resource).to receive(:accessible?).and_return(true)
+    end
+  end
```

Comment:
After talking to Mikhail about this issue I'll retract my earlier comment. The clarity of this spec will improve if we use an `instance_double` for `VSphereCloud::Resources::Datastore` here as we can drop several of the magic values we're using and limit the scope of the unit test to the actual `VMPlacementSelectionPipeline` unit.

Reply:
I see multiple reasons for changing. To be consistent with unit test approach. Where we don't need to rely on what's going on in datastore, we just need to mock it.
Another thing, using instance double shows explicitly what fields you are using. Right now I see 2 params that don't make sense to me: true and 8192. 
The same goes to `fake_placement`.

Reply:
I have another thing to add here. How can I control behavior of all datastores that I am using with just one line. For Instance, if I use Resources instead of instance_double I can say something like 
```
allow_any_instance_of(VSphereCloud::Resources::Datastore).to receive(:maintenance_mode?).and_return(true)
```
I cannot do this on instance_double or is there a way?

Reply:
Since I am dealing with multiple datastores here, I want to control their behavior in bulk. Does this make sense?